### ISSUE TITLE
Feature Request: Add subdirectory support for single domain setup

### ISSUE THREAD

**Issue by adnion** [CollaboraOnline/Docker-CODE#60]
Just wanted to know why users need a subdomain or a second domain for the Collabora Online server to work.

For example cloud.nextcloud.com is the Nextcloud home and Collabora should be served at **cloud.nextcloud.com/office instead of office.nextcloud.com**
I tried a subdirectory setup by changing loolwsd.xml
- file_server_root_path
- service_root
but it didn't work. 
I get 404 because file_server_root_path seems to have no effect.
Thank you for reading.

**Comment by fowlerovski** [comment]
bump

**Comment by kaktuspalme** [comment]
I just added those lines from the documentation to my nextcloud virtualhost config in apache 2:
`ProxyPreserveHost On

# static html, js, images, etc. served from loolwsd
# loleaflet is the client part of LibreOffice Online
ProxyPass           /loleaflet https://127.0.0.1:9980/loleaflet retry=0
ProxyPassReverse    /loleaflet https://127.0.0.1:9980/loleaflet

# WOPI discovery URL
ProxyPass           /hosting/discovery https://127.0.0.1:9980/hosting/discovery retry=0
ProxyPassReverse    /hosting/discovery https://127.0.0.1:9980/hosting/discovery

# Main websocket
ProxyPassMatch "/lool/(.*)/ws$" wss://127.0.0.1:9980/lool/$1/ws nocanon

# Admin Console websocket
ProxyPass   /lool/adminws wss://127.0.0.1:9980/lool/adminws

# Download as, Fullscreen presentation and Image upload operations
ProxyPass           /lool https://127.0.0.1:9980/lool
ProxyPassReverse    /lool https://127.0.0.1:9980/lool

# Endpoint with information about availability of various features
ProxyPass           /hosting/capabilities https://127.0.0.1:9980/hosting/capabilities retry=0
ProxyPassReverse    /hosting/capabilities https://127.0.0.1:9980/hosting/capabilities
`
For nginx it's probably similiar. In my case it works this way on the same domain as my nextcloud.

**Comment by r2evans** [comment]
@kaktuspalme  this looks as if you have are sharing the root directory of the web server between CODE and nextcloud, meaning that (for instance)

```
https://example.com/apps/files           --> nextcloud
https://example.com/lool/.*/ws           --> code
https://example.com/settings/user        --> nextcloud
https://example.com/hosting/capabilities --> code
```

This doesn't seem like it's running CODE in a sub-dir, just banking on the current state that nextcloud and CODE don't use the same top-level directories/end-points. Is that right?

(I'm not knocking the premise, it seems legit, though I'm curious what issues could arise, either within nextcloud or CODE (conflicts) or performance issues in the rev-proxy (whether apache2, nginx, or traefik or similar). Thanks!

**Comment by kaktuspalme** [comment]
@r2evans Yes all ProxyPass paths are served from code within docker and all other paths are served from nextcloud. If one of the code paths would be served from nextcloud you couldn't reach that nextcloud path. So it only works as long as you don't have mutual paths.
